envoy/access_log/access_log.h

```diff
@@ -104,5 +104,8 @@ using FilterPtr = std::unique_ptr<Filter>;
 using Instance = InstanceBase<Formatter::HttpFormatterContext>;
 using InstanceSharedPtr = std::shared_ptr<Instance>;
 
+using InstanceInlinedVector = absl::InlinedVector<InstanceSharedPtr, 4>;
```

Comment:
I suggest not using an inlined vector here (the number 4 doesn't seem to be justified), but a regular vector with an initial reservation. LMK if you think there's a non-negligible impact on used memory which favors the inlined version.

Reply:
The `InstanceInlinedVector` is used only for the quic where we needn't to copy the logger list to the stream for defered stats.
Considering we ususally only have very limit loggers, the inlined vector could be used to avoid unnecessary heap allocation. 

I have no strong point to this because we basically never use the quic.